pkg/provisioner/ironic/ironic.go

```diff
@@ -414,6 +425,27 @@ func (p *ironicProvisioner) Provision(userData string) (result provisioner.Resul
 
 	result.RequeueAfter = provisionRequeueDelay
 
+	// FIXME(dhellmann): The Stein version of Ironic supports passing
+	// a URL. When we upgrade, we can stop doing this work ourself.
+	checksum := p.host.Spec.Image.Checksum
+	isURL, err := checksumIsURL(checksum)
+	if err != nil {
+		return result, errors.Wrap(err, "Could not understand image checksum")
+	}
+	if isURL {
+		p.log.Info("looking for checksum for image", "URL", checksum)
+		resp, err := http.Get(checksum)
+		if err != nil {
+			return result, errors.Wrap(err, "Could not fetch image checksum")
+		}
+		defer resp.Body.Close()
+		checksumBody, err := ioutil.ReadAll(resp.Body)
```

Comment:
I know this code only was relocated and isn't new, but I just noticed that we need to check the response code. Non-2xx codes won't result in an error.

Reply:
Good catch. I pushed a fixup patch to address that.